bundle/permissions/workspace_root.go

```diff
@@ -52,16 +53,60 @@ func giveAccessForWorkspaceRoot(ctx context.Context, b *bundle.Bundle) error {
 	}
 
 	w := b.WorkspaceClient().Workspace
-	obj, err := w.GetStatusByPath(ctx, b.Config.Workspace.RootPath)
+	err := setPermissions(ctx, w, b.Config.Workspace.RootPath, permissions)
 	if err != nil {
 		return err
 	}
 
-	_, err = w.UpdatePermissions(ctx, workspace.WorkspaceObjectPermissionsRequest{
+	// Adding backslash to the root path
+	rootPath := b.Config.Workspace.RootPath
+	if rootPath[len(rootPath)-1] != '/' {
+		rootPath += "/"
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.ArtifactPath, rootPath) {
+		err = setPermissions(ctx, w, b.Config.Workspace.ArtifactPath, permissions)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.FilePath, rootPath) {
+		err = setPermissions(ctx, w, b.Config.Workspace.FilePath, permissions)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.StatePath, rootPath) {
+		err = setPermissions(ctx, w, b.Config.Workspace.StatePath, permissions)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !strings.HasPrefix(b.Config.Workspace.ResourcePath, rootPath) {
+		err = setPermissions(ctx, w, b.Config.Workspace.ResourcePath, permissions)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
These all need to have the `/Workspace/` prefix as well. At least the artifact path can have `/Volumes/`.